src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+// Disable revive linter since any structs in this file will expose the same
+// funcs as the original structs in the msgraph-sdk-go package, which do not
+// follow some of the golint rules.
+//
+//nolint:revive
+package custom
+
+import (
+	"strings"
+	"time"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+// Replica of models.DriveItemable
```

Comment:
Does this struct need Setter funcs to completely comply with that interface?

Reply:
Clarified in another comment. `LiteDriveItemable` doesn't implement the entire `DriveItemable` interface. Also, even if it did ( function wise), we will have to construct return interfaces (e.g. `ItemReferenceable`, `Malwareable` etc.) to fully comply with `DriveItemable`, which would beat the whole `lite` aspect.